source/routing/raptor_solution_reader.cpp

```diff
@@ -88,7 +88,14 @@ get_out_st_dt(const std::pair<const type::StopTime*, DateTime>& in_st_dt,
     return {nullptr, 0};
 }
 
-bool is_valid(const Journey& j) {
+template<typename Visitor>
+bool is_valid(const RaptorSolutionReader<Visitor>& reader, const Journey& j) {
+    // We don't want journeys that begins before (resp. after) the requested datetime
+    if (reader.v.clockwise()) {
+        if (j.departure_dt < reader.departure_datetime) { return false; }
```

Comment:
you can `return j.departure_dt >= reader.departure_datetime `.  same thing for L97